packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts

```diff
@@ -260,12 +260,49 @@ describe("toMapTree", () => {
 				["b", "Hello world"],
 				["c", null],
 				["d", undefined as unknown as InsertableContent], // Should be skipped in output
-				["e", { age: 37, name: "Jill" }],
 			];
 			const tree = new Map<string, InsertableContent>(entries);
 
 			const actual = nodeDataToMapTree(tree, [schema]);
 
+			const expected: MapTree = {
+				type: brand("test.map"),
+				fields: new Map<FieldKey, MapTree[]>([
+					[brand("a"), [{ type: leaf.number.name, value: 42, fields: new Map() }]],
+					[
+						brand("b"),
+						[{ type: leaf.string.name, value: "Hello world", fields: new Map() }],
+					],
+					[brand("c"), [{ type: brand(leaf.null.name), value: null, fields: new Map() }]],
+				]),
+			};
+
+			assert.deepEqual(actual, expected);
+		});
+
+		it("Non-empty map (input type: Record)", () => {
+			const schemaFactory = new SchemaFactory("test");
+			const childObjectSchema = schemaFactory.object("child-object", {
+				name: schemaFactory.string,
+				age: schemaFactory.number,
+			});
+			const schema = schemaFactory.map("map", [
+				childObjectSchema,
+				schemaFactory.number,
+				schemaFactory.string,
+				schemaFactory.null,
+			]);
+
+			const tree: Record<string, InsertableContent> = {
```

Comment:
the set of types allowed should match thats allowed in the constructor of a map. THis data is not compatible with that since  its not an iterator. Trying touse this with the actual public APIs should not build,

Reply:
For now, I've reverted the functional changes to this code. We can do a follow-up PR to change the input contracts.